src/main/java/frc/lib/team3061/drivetrain/Drivetrain.java

```diff
@@ -470,13 +473,48 @@ public void periodic() {
     // check for teleportation
     if (this.inputs.drivetrain.robotPose.minus(prevRobotPose).getTranslation().getNorm() > 0.4) {
       this.resetPose(prevRobotPose);
+      this.teleportedCount++;
       Logger.recordOutput(SUBSYSTEM_NAME + "/TeleportedPose", this.inputs.drivetrain.robotPose);
-      Logger.recordOutput(SUBSYSTEM_NAME + "/Teleported", true);
+      Logger.recordOutput(SUBSYSTEM_NAME + "/TeleportCount", this.teleportedCount);
     } else {
       this.prevRobotPose = this.inputs.drivetrain.robotPose;
-      Logger.recordOutput(SUBSYSTEM_NAME + "/Teleported", false);
     }
 
+    // check for position outside the field due to slipping
+    if (this.inputs.drivetrain.robotPose.getX() < 0) {
+      this.resetPose(
+          new Pose2d(
+              0,
+              this.inputs.drivetrain.robotPose.getY(),
+              this.inputs.drivetrain.robotPose.getRotation()));
+      this.constrainPoseToFieldCount++;
+    } else if (this.inputs.drivetrain.robotPose.getX() > FieldConstants.fieldLength) {
+      this.resetPose(
+          new Pose2d(
+              FieldConstants.fieldLength,
+              this.inputs.drivetrain.robotPose.getY(),
+              this.inputs.drivetrain.robotPose.getRotation()));
+      this.constrainPoseToFieldCount++;
+    }
+
+    if (this.inputs.drivetrain.robotPose.getY() < 0) {
```

Comment:
I split this into two separate if statements since the pose may leave the field in both x and y directions at the same time.